src/pages/studyView/StudyViewUtils.tsx

```diff
@@ -521,7 +521,9 @@ export function generateNumericalData(numericalBins: DataBin[]): BarDatum[] {
         return {
             x: x + xShift,
             y: dataBin.count,
-            dataBin
+            dataBin,
+            label:(dataBin.start === undefined || dataBin.end === undefined) ? dataBin.start === undefined ? (dataBin.start === undefined && dataBin.end === undefined) ? "No of Samples:"+dataBin.count + " Range:NotDefine": "No of Samples:"+dataBin.count+" "+ "and Range≤" + dataBin.end:"No of Samples:"+dataBin.count+" "+ "and Range≥" + dataBin.start :"No of Samples:"+dataBin.count+" " + "and Range:"+ dataBin.start+"-"+ dataBin.end  
```

Comment:
- Please consolidate the logics to a util function and add tests.
- Try to use template string for string concatenation.
- Typo:  It should be `No.`
- Can we have No. of Samples and Range on separate lines?

Reply:
I didn't understand why some checks were not successful. I didn't get the error. so please can anyone explains what's mistake and error?